code/modules/research/xenobiology/crossbreeding/warping.dm

```diff
@@ -0,0 +1,1135 @@
+
+
+/obj/effect/warped_rune/goldspace
+	icon_state = "midas_rune"
+	desc = "When everything is abundant, it becomes easy to forget when one had to work to obtain anything."
+	max_cooldown = 600
+
+
+/obj/effect/warped_rune/goldspace/Initialize()
+	. = ..()
+	START_PROCESSING(SSprocessing, src)
+
+
+///turn anything on the tile to a golden version of themselves. Sheets of any kind will be directly turned into gold bars.
+/obj/effect/warped_rune/goldspace/process()
+	if(cooldown > world.time)
+		return
+
+	for(var/obj/item/goldened in rune_turf)
+		if(istype(goldened,/obj/item/stack/sheet/mineral/gold))//can't turn gold into gold
+			return
+
+		var/gold_amount = 0
+		for(var/mats in goldened.custom_materials)
+			gold_amount += goldened.custom_materials[mats]
+		if(!gold_amount)
+			gold_amount = 50 //if the item doesn't have any material it makes the item worth around 0.005 gold bars.
+		goldened.material_flags = MATERIAL_COLOR | MATERIAL_ADD_PREFIX | MATERIAL_AFFECT_STATISTICS
+		goldened.set_custom_materials(list(/datum/material/gold = gold_amount))
+		cooldown = world.time + max_cooldown //only set the cooldown once something's been turned to gold
+		return
+
+
+/*Adamantine rune, will spawn ores depending on the mineral rocks surrounding it. Here to make miners do their job even less.  */
+
+
+/obj/item/slimecross/warping/adamantine
+	colour = "adamantine"
+	runepath = /obj/effect/warped_rune/adamantinespace
+	effect_desc = "draws a rune capable of copying the ores of nearby mineral rocks."
+
+
+/obj/effect/warped_rune/adamantinespace  //doesn't have a rune icon yet please spriters help me I can't sprite for shit I beg you
+	desc = "The universe's ressource are nothing but tools for us to use and abuse."
+	max_cooldown = 300  //"mines" things every 30 seconds.
+	icon_state = "mining_rune"
+
+
+/obj/effect/warped_rune/adamantinespace/Initialize()
+	. = ..()
+	START_PROCESSING(SSprocessing, src)
+
+
+/obj/effect/warped_rune/adamantinespace/process()
+	if(cooldown < world.time)
+		cooldown = world.time + max_cooldown
+		auto_mining()
+
+
+/obj/effect/warped_rune/adamantinespace/proc/auto_mining()
+	for(var/turf/closed/mineral/ores in range(7,rune_turf)) //the range is pretty big to at least try to rival miners and their plasma cutters.
+		if(ores.mineralType != null) //here to counter runtimes when the mineral type of the rock is null
+			new ores.mineralType(rune_turf)
+
+
+/* Lightpink rune. Revive suicided/soulless corpses by yeeting a willing soul into it */
+
+
+/obj/item/slimecross/warping/lightpink
+	colour = "light pink"
+	runepath = /obj/effect/warped_rune/lightpinkspace
+	effect_desc = "draws a rune that will repair the soul of a humanoid corpse in the hope of bringing them back to life."
+	drawing_time = 100
+
+
+/obj/effect/warped_rune/lightpinkspace
+	icon_state = "necro_rune"
+	desc = "Souls are like any other material, You just have to find the right place to manufacture them."
+	max_cooldown = 100
```

Comment:
might want to increase this cooldown to, like, a minute, to prevent ghost ping spam (and to keep xenobio from being able to pump out living humanoids TOO quickly)